golang/lib/execution/test_suite_executor.go

```diff
@@ -157,41 +154,31 @@ func runTestExecutionFlow(ctx context.Context, testsuite testsuite.TestSuite, co
 		executionClient,
 		filesArtifactUrls)
 
-	// TODO Also time out the setup with the API container rather than storing this locally
-	//  to reduce complexity inside the lib
 	logrus.Info("Setting up the test network...")
 	untypedNetwork, err := test.Setup(networkCtx)
 	if err != nil {
 		return stacktrace.Propagate(err, "An error occurred setting up the test network")
 	}
+	time.Sleep(10 * time.Second)
+	if _, err := executionClient.RegisterTestSetupCompletion(ctx, &emptypb.Empty{}); err != nil {
+		return stacktrace.Propagate(err, "An error occurred registering the test setup completion with the API container")
+	}
 	logrus.Info("Test network set up")
 
 	logrus.Infof("Executing test '%v'...", testName)
+
+	if _, err := executionClient.RegisterTestExecution(ctx, &emptypb.Empty{}); err != nil {
+		return stacktrace.Propagate(err, "An error occurred registering the test execution with the API container")
+	}
 	testResultChan := make(chan error)
 
 	go func() {
 		testResultChan <- runTestInGoroutine(test, untypedNetwork)
 	}()
 
-	// TODO Switch to registering the timeout with the API container rather than storing this locally
-	//  to reduce complexity inside the lib
-	// Time out the test so a poorly-written test doesn't run forever
-	testTimeout := test.GetExecutionTimeout()
-	var timedOut bool
-	var testResultErr error
-	select {
-	case testResultErr = <- testResultChan:
-		logrus.Tracef("Test returned result before timeout: %v", testResultErr)
-		timedOut = false
-	case <- time.After(testTimeout):
-		logrus.Tracef("Hit timeout %v before getting a result from the test", testTimeout)
-		timedOut = true
-	}
-	logrus.Tracef("After running test w/timeout: resultErr: %v, timedOut: %v", testResultErr, timedOut)
+	testResultErr := <- testResultChan
```

Comment:
No more channels needed now!

Reply:
same comment as above...my understanding is that we get nice error checking on a panic in test execution with this gofunc technique, but I could be totally off here. lmk if we'd miss out by making this change